pex/variables.py

```diff
@@ -34,14 +34,26 @@ def iter_help(cls):
       yield variable_name, variable_type, variable_text
 
   @classmethod
-  def from_rc(cls, rc):
+  def from_rc(cls, rc, build_time_rc_dir=''):
     """Read pex runtime configuration variables from a pexrc file.
 
-       :param rc: an absolute path to a pexrc file
-       :return ret_vars: a dict object containing key values pairs found in processed pexrc files
+    :param rc: an absolute path to a pexrc file
+    :param build_time_rc_dir: an absolute path to a directory containing a pexrc. This is needed to
+    read a pexrc at build time in the case where a pexrc is located in the same directory as a the
+    output pex.
+
+    :return ret_vars: a dict object containing key values pairs found in processed pexrc files
     """
     ret_vars = {}
-    for filename in ['/etc/pexrc', rc, os.path.join(os.path.dirname(sys.argv[0]), '.pexrc')]:
+    rc_locations = ['/etc/pexrc',
+                    rc,
+                    os.path.join(os.path.dirname(sys.argv[0]), '.pexrc'),
+                    os.getcwd()]
+    if not rc:
+      rc_locations.append('~/.pexrc')
```

Comment:
The only time `~/.pexrc` is the default is if we are calling the Variables constructor; the `from_rc` api requires a string so I added support for adding `~/.pexrc` if the input is blank. I suppose I could make it a default arg or pass it in in PEX CLI...

Reply:
Added it as default arg.

Reply:
seems to me like `~/.pexrc` needs to be considered in all cases (i.e. as a static member of `rc_locations`) or else we aren't reading from all of the places that pexrc configuration can exist. (essentially, I don't see any existing supporting usage of `Variables` in pex/pants to warrant keeping the `rc` param here the way it is).